src/main/java/com/microsoft/lst_bench/client/JDBCConnection.java

```diff
@@ -43,6 +52,27 @@ public void execute(String sqlText) throws ClientException {
     }
   }
 
+  @Override
+  public Object executeQuery(String sqlText) throws ClientException {
+    List<Map<String, Object>> value_list = new ArrayList<>();
+    try (Statement s = connection.createStatement()) {
+      ResultSet rs = s.executeQuery(sqlText);
+      ResultSetMetaData rsmd = rs.getMetaData();
+
+      while (rs.next()) {
```

Comment:
If the `QueryResult` object is created, it could be implemented as an iterator and this retrieval could be done lazily, so we do not need to hold the full query result in memory.

Reply:
I've tried something like that using a CachedRowSet but I've gotten a function error, my guess is that the JDBC driver didn't have an appropriate population function (CachedRowSet.populate(ResultSet)). ResultSet is not persisted outside of the Statement which means that as soon as the Statement closes, the ResultSet will be closed as well, at least that's what I found while testing this approach. So we do have to persist this within this loop somehow but I'm happy about populating a new QueryResult object that has a better-defined API. What do you think?

Reply:
Added an implementation for QueryResult, let me know what you think.